pykern/pkcli/github.py

```diff
@@ -49,6 +50,44 @@ def backup():
     pkdlog("DONE")
 
 
+def ci_check(repo, branch=None):
+    def _branch(r, name, reraise=True):
+        try:
+            return r.branch(name=name)
+        except github3.exceptions.NotFoundError:
+            if reraise:
+                raise
+
+    def _run_on_repo(*cmd):
+        return (
+            subprocess.check_output(
+                ["git", "-C", f"{os.environ['HOME']}/src/{repo}", *cmd],
+                stderr=subprocess.STDOUT,
+            )
+            .decode("utf-8")
+            .rstrip("\n")
+        )
+
+    r = _repo_arg(repo)
+    b = (
+        _branch(r, branch)
+        if branch
+        else _branch(r, "master", False) or _branch(r, "main")
+    )
+    s = b.commit.sha
+    _run_on_repo("checkout", b.name)
+    _run_on_repo("pull")
+    o = _run_on_repo("rev-parse", f"origin/{b.name}")
+    if not s == o:
+        raise pkcli.command_error(f"Most recent run on {b.name}={s} does not match {o}")
+    c = [c.conclusion for c in b.commit.check_runs()]
+    if not c:
+        raise pkcli.command_error("No workflow runs for commit")
+    if "success" not in c or any(x not in ("success", "skipped") for x in c):
```

Comment:
I think you need to verify that the most recent run is successful. I don't know really what the content of check_runs can be, but I'm guessing if you rerun a check (for transient errors, for example), then it will add more runs.

Reply:
What is it skipping? That is, are there two workflows and it is skipping one because it doesn't match the criteria? The reason I'm asking is the test is rather complicated. It would be easier to have: c[0] == "success" or something like that.

Reply:
Yes, I was testing on a branch that had two jobs in the same workflow. If we'll only ever have one job per workflow then  c[0] == "success" makes sense. 
In checking this I realized that the change to the workflow file needs to be made in sirepo too, and I never changed it in sirepo when I fixed #223. I'll open a pr for that as well.

Reply:
92f8cb5ae798c6456ced913c26ada41747101e27